L1Trigger/TrackFindingTMTT/src/MuxHToutputs.cc

```diff
@@ -0,0 +1,192 @@
+
+//--- Note that the word "link" appearing in the C++ or comments in this class actually corresponds
+//--- to a pair of links in the hardware.
+
+#include "L1Trigger/TrackFindingTMTT/interface/MuxHToutputs.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Settings.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+
+namespace TMTT {
+
+  //=== Initialize constants from configuration parameters.
+
+  MuxHToutputs::MuxHToutputs(const Settings* settings)
+      : settings_(settings),
+        muxOutputsHT_(settings_->muxOutputsHT()),
+        numPhiNonants_(settings_->numPhiNonants()),
+        numPhiSectors_(settings_->numPhiSectors()),
+        numPhiSecPerNon_(numPhiSectors_ / numPhiNonants_),
+        numEtaRegions_(settings_->numEtaRegions()),
+        busySectorKill_(settings_->busySectorKill()),          // Kill excess tracks flowing out of HT?
+        busySectorNumStubs_(settings_->busySectorNumStubs()),  // Max. num. of stubs that can be sent within TM period
+        busySectorMbinRanges_(
+            settings_->busySectorMbinRanges()),  // Individual m bin (=q/Pt) ranges to be output to opto-links.
+        busySectorUseMbinRanges_(busySectorMbinRanges_.size() > 0)  // m bin ranges option disabled if vector empty.
+  {
+    // Implemented MUX algorithm relies on same number of sectors per nonant.
+    if (numPhiSectors_ % numPhiNonants_ != 0)
+      throw cms::Exception("MuxHToutputs: Number of phi sectors is not a multiple of number of nonants!");
+
+    if (!busySectorUseMbinRanges_)
+      throw cms::Exception(
+          "MuxHToutputs: The implemented MUX algorithm requires you to be using the busySectorMbinRanges cfg option!");
+
+    // Check that the MUX algorithm implemented in linkID() is not obviously wrong.
+    this->sanityCheck();
+
+    bool static first = true;
```

Comment:
Please use `std::call_once()`.

Reply:
Am now using call_once() for all debug messages that need printing only once.